lib/rouge/lexers/cpp.rb

```diff
@@ -57,9 +57,11 @@ def self.reserved
 
       prepend :statements do
         rule /class\b/, Keyword, :classname
-        rule %r((#{dq}[.]#{dq}?|[.]#{dq})(e[+-]?#{dq}[lu]*)?)i, Num::Float
-        rule %r(#{dq}e[+-]?#{dq}[lu]*)i, Num::Float
+        rule %r((#{dq}[.]#{dq}?|[.]#{dq})(e[+-]?#{dq}[lf]?)?)i, Num::Float
```

Comment:
From the doc, 

suffix, if present, is one of f, F, l, or L. The suffix determines the type of the floating-point literal:
 - (no suffix) defines double
  - f F defines float
 - l L defines long double 

So I changed `[lu]*` for `[lf]?`